core/domain/learner_progress_services.py

```diff
@@ -420,3 +470,46 @@ def get_all_incomplete_collection_ids(user_id):
         return incomplete_activities.collection_ids
     else:
         return []
+
+
+def remove_none_entities_from_list(
+        user_id, entity_summary_list, entity_ids, section, sub_section):
```

Comment:
sub_section --> subsection

I feel somewhat iffy about this function; it has serious side-effects that are not conveyed well in its name. Perhaps consider doing something like this instead:

    update_and_retrieve_user_progress(user_id):
        # Bulk-retrieve user progress data (using ndb.get_multi())
        # If needed, process removals and bulk-put() any models which have changed
        # Return up-to-date user progress data

Then you get two datastore calls per-get which seems reasonable. Not sure how many you have currently...